### DISCUSSION TITLE
Help - new docker user

### DISCUSSION THREAD

**Discussion by RewindRep** [ep1cman/unifi-protect-backup#87] [Q&A]
Hi, I think this is an amazing project but I am new to docker and rclone. have you considered a step by step guide for the first time docked users, as this would be super helpful to me?

**Comment by tipplad** [comment]
I second this, a step by step guide / video would be great.

**Comment by ep1cman** [comment]
Which parts of the setup are you specifically having difficulties with? 

Docker installation guides can be found here: https://docs.docker.com/get-docker/
(if you are using linux, I would recommend docker engine instead of docker desktop: https://docs.docker.com/engine/install/)

For setting up `rclone` this project's readme has the following command:
```
$ docker run -it --rm -v $PWD:/root/.config/rclone --entrypoint rclone ghcr.io/ep1cman/unifi-protect-backup config
```
This will take you through an interactive setup to create your `rclone.conf` file.

**Reply by RewindRep** [reply]
Thanks for the response, this is a totally new world to me! I have docker installed and have tried to run the rclone.cfg setup but I am unsure where the file goes and how then to import it to the main docker container, if that makes sense?

**Reply by ep1cman** [reply]
It should appear in the directory you ran the command. What OS are you using and how are you launching the container? (Command line, docker desktop etc).

**Reply by RewindRep** [reply]
I’m using Win 10 and launching through command line

**Reply by RewindRep** [reply]
Also trying to backup to back blaze B2 but am open to using anything else if you can help troubleshoot easier

**Reply by ep1cman** [reply]
Ah ok, the instructions assume Linux/MacOS. The command above wont work on windows because the `$PWD` variable wont exists. From some googing I believe it should be "%CD%" so the command is:

```
docker run -it --rm -v %CD%:/root/.config/rclone --entrypoint rclone ghcr.io/ep1cman/unifi-protect-backup config
```

Let me know if this works, and I will update the docs accordingly

**Reply by billmallard** [reply]
This worked for me in Windows running Rancher Desktop as my Docker & Kubernetes platform.  

Depending on where you run the command - if it's default command prompt location it will drop the conf file in the C:\Users\YOURUSERNAME\rclone.conf location.

**Comment by tipplad** [comment]
Hi Sebastain,

Thanks for reaching out, sorry, it's a bit new to me, when creating my own
client ID (following the instructions here
<https://rclone.org/drive/#making-your-own-client-id>), under step 5 it
says "If you already configured an "Oauth Consent Screen", then skip to the
next step; if not, click on "CONFIGURE CONSENT SCREEN" button (near the top
right corner of the right panel), then select "External" and click on
"CREATE";"

I however don't see this option:

[image: image.png]

Is there a discord channel I can get some help from? I don't want to be
bothering you with stupid questions like this!

Thanks very much for the project, it's exactly what i've been looking for
(if I can get it to work)!

On Wed, 26 Apr 2023 at 10:29, Sebastian Goscik ***@***.***>
wrote:

> Which parts of the setup are you specifically having difficulties with?
>
> Docker installation guides can be found here:
> https://docs.docker.com/get-docker/
> (if you are using linux, I would recommend docker engine instead of docker
> desktop: https://docs.docker.com/engine/install/)
>
> For setting up rclone this project's readme has the following command:
>
> $ docker run -it --rm -v $PWD:/root/.config/rclone --entrypoint rclone ghcr.io/ep1cman/unifi-protect-backup config
>
> This will take you through an interactive setup to create your rclone.conf
> file.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ep1cman/unifi-protect-backup/discussions/87#discussioncomment-5729288>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZTRF5A5YKVY2JAGPCHRJGTXDDTJJANCNFSM6AAAAAAXKBPTYI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
David Ronan

**Reply by ep1cman** [reply]
Unfortunately I am unable to help with this since I am not familiar with all the cloud storage backends provided by `rclone` (I don't use Google Drive).

I would recommend asking on the rclone forums: https://forum.rclone.org/
